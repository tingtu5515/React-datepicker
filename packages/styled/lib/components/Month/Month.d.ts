import {FirstDayOfWeek} from '@datepicker-react/hooks'
interface MonthProps {
  year: number
  month: number
  firstDayOfWeek: FirstDayOfWeek
  dayFormat(date: Date): string
  weekDayFormat(date: Date): string
  monthLabelFormat(date: Date): string
}
declare const Month: ({
  year,
  month,
  firstDayOfWeek,
  dayFormat,
  monthLabelFormat,
  weekDayFormat,
}: MonthProps) => JSX.Element
export default Month
