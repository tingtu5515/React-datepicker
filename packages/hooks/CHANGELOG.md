# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.6.0](https://github.com/tresko/react-datepicker/compare/v1.5.0...v1.6.0) (2019-07-14)


### Bug Fixes

* **styled-system:** upgrade to version 5.0.12 ([6ce33b1](https://github.com/tresko/react-datepicker/commit/6ce33b1))






## [1.5.1](https://github.com/tresko/react-datepicker/compare/v1.5.0...v1.5.1) (2019-07-01)


### Bug Fixes

* **styled-system:** upgrade to version 5.0.12 ([6ce33b1](https://github.com/tresko/react-datepicker/commit/6ce33b1))





## [1.4.1](https://github.com/tresko/react-datepicker/compare/v1.4.0...v1.4.1) (2019-06-19)


### Bug Fixes

* **accessibility:** accessibility fix ([2779136](https://github.com/tresko/react-datepicker/commit/2779136))





# [1.3.0](https://github.com/tresko/react-datepicker/compare/v1.2.3-alpha.3...v1.3.0) (2019-06-11)

**Note:** Version bump only for package @datepicker-react/hooks





## [1.2.3-alpha.3](https://github.com/tresko/react-datepicker/compare/v1.2.3-alpha.2...v1.2.3-alpha.3) (2019-06-05)


### Bug Fixes

* **styled:** roll back styled system ([6990308](https://github.com/tresko/react-datepicker/commit/6990308))





# 1.2.0 (2019-04-04)


### Bug Fixes

* **datepicker:** fix numberOfMonths ([a287c0e](https://github.com/tresko/react-datepicker/commit/a287c0e))
* **scope:** change packages scope ([2a83211](https://github.com/tresko/react-datepicker/commit/2a83211))
* **test:** test ([190bced](https://github.com/tresko/react-datepicker/commit/190bced))
* **versions:** versions ([3219781](https://github.com/tresko/react-datepicker/commit/3219781))


### Features

* **release:** first canary ([aedf88c](https://github.com/tresko/react-datepicker/commit/aedf88c))





# 1.1.0 (2019-04-03)


### Bug Fixes

* **datepicker:** fix numberOfMonths ([a287c0e](https://github.com/tresko/react-datepicker/commit/a287c0e))
* **scope:** change packages scope ([2a83211](https://github.com/tresko/react-datepicker/commit/2a83211))
* **test:** test ([190bced](https://github.com/tresko/react-datepicker/commit/190bced))
* **versions:** versions ([3219781](https://github.com/tresko/react-datepicker/commit/3219781))


### Features

* **release:** first canary ([aedf88c](https://github.com/tresko/react-datepicker/commit/aedf88c))
