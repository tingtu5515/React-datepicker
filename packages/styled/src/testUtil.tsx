import React from 'react'
import {render} from '@testing-library/react'
import DatepickerContext, {datepickerContextDefaultValue} from './context/datepickerContext'

function renderWithProviders(ui: any, options?: any, contextValues?: Record<string, unknown>) {
  return render(
    <DatepickerContext.Provider value={{...datepickerContextDefaultValue, ...contextValues}}>
      {ui}
    </DatepickerContext.Provider>,
    options,
  )
}

export * from '@testing-library/react'
export {renderWithProviders as render}
