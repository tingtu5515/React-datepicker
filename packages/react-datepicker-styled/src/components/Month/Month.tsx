import React from 'react'
import {useMonth, CalendarDay} from '@react-datepicker/hooks'
import MonthLabel from '../MonthLabel'
import DayLabel from '../DayLabel'
import Flex from '../Flex'
import Grid from '../Grid'
import Day from '../Day'

interface MonthProps {
  year: number
  month: number
}

const Month = ({year, month}: MonthProps) => {
  const {days, weekDays, monthLabel} = useMonth({year, month})

  return (
    <div>
      <Flex justifyContent="center" mb="28px">
        <MonthLabel label={monthLabel} />
      </Flex>
      <Grid gridTemplateColumns="repeat(7, 36px)">
        {weekDays.map(day => (
          <Flex key={day} justifyContent="center" mb="16px">
            <DayLabel label={day} />
          </Flex>
        ))}
      </Grid>
      <Grid gridTemplateColumns="repeat(7, 36px)">
        {days.map((day: CalendarDay, index: number) => {
          if (typeof day === 'object') {
            return (
              <Day
                isActive={false}
                key={day.day}
                day={day.day}
                disabled={false}
                isStartOrEnd={false}
              />
            )
          }
          return <div key={index} />
        })}
      </Grid>
    </div>
  )
}

export default Month
