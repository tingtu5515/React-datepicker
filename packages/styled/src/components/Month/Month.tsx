import React from 'react'
import {useMonth, CalendarDay} from '@datepicker-react/hooks'
import MonthLabel from '../MonthLabel'
import DayLabel from '../DayLabel'
import Flex from '../Flex'
import Grid from '../Grid'
import Day from '../Day'
import useThemeProps from '../../hooks/useThemeProps'
import globalStyles from '../../globalStyles'
// eslint-disable-next-line import/no-unresolved
import {MonthTheme} from '../../@types/theme'

interface MonthProps {
  year: number
  month: number
  firstDayOfWeek: 0 | 1 | 2 | 3 | 4 | 5 | 6
  isDateBlocked(date: Date): boolean
  isDateSelected(date: Date): boolean
  isStartOrEndDate(date: Date): boolean
  isDateHovered(date: Date): boolean
  onDaySelect(date: Date): void
  onDayHover(date: Date): void
}

const Month = ({
  year,
  month,
  firstDayOfWeek,
  isDateBlocked,
  isDateSelected,
  isStartOrEndDate,
  onDaySelect,
  onDayHover,
  isDateHovered,
}: MonthProps) => {
  const {days, weekDays, monthLabel} = useMonth({year, month, weekStartsOn: firstDayOfWeek})
  const theme: MonthTheme = useThemeProps({
    daySize: globalStyles.daySize,
    monthLabelMargin: '0 0 28px',
    monthDayLabelMargin: '0 0 16px',
  })

  return (
    <div>
      <Flex justifyContent="center" m={theme.monthLabelMargin}>
        <MonthLabel label={monthLabel} />
      </Flex>
      <Grid daySizeGridTemplateColumns={theme.daySize}>
        {weekDays.map((day: string) => (
          <Flex key={day} justifyContent="center" m={theme.monthDayLabelMargin}>
            <DayLabel label={day} />
          </Flex>
        ))}
      </Grid>
      <Grid daySizeGridTemplateColumns={theme.daySize}>
        {days.map((day: CalendarDay, index: number) => {
          if (typeof day === 'object') {
            return (
              <Day
                isActive={isDateSelected(day.date)}
                date={day.date}
                key={day.day}
                day={day.day}
                disabled={isDateBlocked(day.date)}
                isStartOrEnd={isStartOrEndDate(day.date)}
                onDaySelect={onDaySelect}
                onDayHover={onDayHover}
                isWithinHoverRange={isDateHovered(day.date)}
              />
            )
          }
          return <div key={index} />
        })}
      </Grid>
    </div>
  )
}

export default Month
