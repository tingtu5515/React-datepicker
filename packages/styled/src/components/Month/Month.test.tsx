import * as React from 'react'
import {dayFormat, weekDayFormat, monthLabelFormat} from '@datepicker-react/hooks'
import {render} from '../../testUtil'
import Month from '.'

test('should match snapshot', () => {
  const {container, getByText, getAllByTestId} = render(
    <Month
      month={1}
      year={2019}
      firstDayOfWeek={1}
      dayFormat={dayFormat}
      weekDayFormat={weekDayFormat}
      monthLabelFormat={monthLabelFormat}
    />,
  )
  expect(container).toMatchSnapshot()
  getByText('February 2019')

  const days = getAllByTestId('Day')
  expect(days[0].textContent).toBe('01')
  expect(days[days.length - 1].textContent).toBe('28')

  const dayLabels = getAllByTestId('DayLabel')
  expect(dayLabels[0].textContent).toBe('Mo')
  expect(dayLabels[dayLabels.length - 1].textContent).toBe('Su')
})

test('Sunday should be first day of the week', () => {
  const {container, getByText, getAllByTestId} = render(
    <Month
      month={1}
      year={2019}
      firstDayOfWeek={0}
      dayFormat={dayFormat}
      weekDayFormat={weekDayFormat}
      monthLabelFormat={monthLabelFormat}
    />,
  )
  expect(container).toMatchSnapshot()
  getByText('February 2019')

  const days = getAllByTestId('Day')
  expect(days[0].textContent).toBe('01')
  expect(days[days.length - 1].textContent).toBe('28')

  const dayLabels = getAllByTestId('DayLabel')
  expect(dayLabels[0].textContent).toBe('Su')
  expect(dayLabels[dayLabels.length - 1].textContent).toBe('Sa')
})
