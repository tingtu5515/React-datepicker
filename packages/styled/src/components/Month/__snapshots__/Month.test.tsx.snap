// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sunday should be first day of the week 1`] = `
.c1 {
  font-family: Montserrat,sans-serif;
  font-size: 14px;
  font-weight: 600;
  color: #343132;
  line-height: 1.57;
}

.c4 {
  font-family: Montserrat,sans-serif;
  font-size: 11px;
  font-weight: 500;
  color: #929598;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 28px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  width: 100%;
}

.c2 {
  display: grid;
  grid-template-columns: repeat(7,36px);
}

.c5 {
  height: 36px;
  width: 36px;
  box-shadow: 1px 0 0 0 #e6e7e8, 0 1px 0 0 #e6e7e8, 1px 1px 0 0 #e6e7e8, 1px 0 0 0 #e6e7e8 inset, 0 1px 0 0 #e6e7e8 inset;
  background: #ffffff;
  color: #58595b;
  font-family: Montserrat,sans-serif;
  font-weight: 500;
  font-size: 14px;
  cursor: pointer;
  border: 0;
  padding: 0;
  outline: 0;
}

.c5:hover {
  background: #e6e7e8;
  color: #58595b;
}

.c5:focus {
  box-shadow: none;
  border: 2px solid #009fef;
}

<div>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
        color="#343132"
        data-testid="MonthLabel"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="600"
      >
        February 2019
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Su
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Mo
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Tu
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          We
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Th
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Fr
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Sa
        </div>
      </div>
    </div>
    <div
      class="c2"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          01
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          02
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          03
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          04
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          05
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          06
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          07
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          08
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          09
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          10
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          11
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          12
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          13
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          14
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          15
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          16
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          17
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          18
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          19
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          20
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          21
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          22
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          23
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          24
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          25
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          26
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          27
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          28
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`should match snapshot 1`] = `
.c1 {
  font-family: Montserrat,sans-serif;
  font-size: 14px;
  font-weight: 600;
  color: #343132;
  line-height: 1.57;
}

.c4 {
  font-family: Montserrat,sans-serif;
  font-size: 11px;
  font-weight: 500;
  color: #929598;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 28px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  width: 100%;
}

.c2 {
  display: grid;
  grid-template-columns: repeat(7,36px);
}

.c5 {
  height: 36px;
  width: 36px;
  box-shadow: 1px 0 0 0 #e6e7e8, 0 1px 0 0 #e6e7e8, 1px 1px 0 0 #e6e7e8, 1px 0 0 0 #e6e7e8 inset, 0 1px 0 0 #e6e7e8 inset;
  background: #ffffff;
  color: #58595b;
  font-family: Montserrat,sans-serif;
  font-weight: 500;
  font-size: 14px;
  cursor: pointer;
  border: 0;
  padding: 0;
  outline: 0;
}

.c5:hover {
  background: #e6e7e8;
  color: #58595b;
}

.c5:focus {
  box-shadow: none;
  border: 2px solid #009fef;
}

<div>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
        color="#343132"
        data-testid="MonthLabel"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="600"
      >
        February 2019
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Mo
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Tu
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          We
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Th
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Fr
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Sa
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
          color="#929598"
          data-testid="DayLabel"
          font-family="Montserrat, sans-serif"
          font-size="11px"
          font-weight="500"
        >
          Su
        </div>
      </div>
    </div>
    <div
      class="c2"
    >
      <div />
      <div />
      <div />
      <div />
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          01
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          02
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          03
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          04
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          05
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          06
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          07
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          08
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          09
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          10
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          11
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          12
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          13
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          14
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          15
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          16
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          17
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          18
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          19
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          20
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          21
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          22
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          23
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          24
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          25
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          26
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          27
        </div>
      </button>
      <button
        class="c5"
        color="#58595b"
        data-testid="Day"
        font-family="Montserrat, sans-serif"
        font-size="14px"
        font-weight="500"
        role="button"
      >
        <div
          class="c6"
          height="100%"
          width="100%"
        >
          28
        </div>
      </button>
    </div>
  </div>
</div>
`;
